ompi/mca/osc/ucx/osc_ucx_active_target.c

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2017. ALL RIGHTS RESERVED.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+#include "ompi_config.h"
+
+#include "ompi/mca/osc/osc.h"
+#include "ompi/mca/osc/base/base.h"
+#include "ompi/mca/osc/base/osc_base_obj_convert.h"
+
+#include "osc_ucx.h"
+
+
+int ompi_osc_ucx_fence(int assert, struct ompi_win_t *win) {
+    ompi_osc_ucx_module_t *module = (ompi_osc_ucx_module_t*) win->w_osc_module;
+    int ret = OMPI_SUCCESS;
+    ucs_status_t status;
+
+    if (module->epoch_type.access != NONE_EPOCH &&
+        module->epoch_type.access != FENCE_EPOCH) {
+        return OMPI_ERR_RMA_SYNC;
+    }
+
+    if (assert & MPI_MODE_NOSUCCEED) {
+        module->epoch_type.access = NONE_EPOCH;
+    } else {
+        module->epoch_type.access = FENCE_EPOCH;
+    }
```

Comment:
(now L32)
I think this is the same problem with the first long discussion. When the user call Fence with no assert, MPI library cannot decide if epoch is closed or not, and the state should always be FENCE_EPOCH.
So, I think it is meaningless to set exposure epoch with FENCE. The epoch is left to be FENCE until it meets POST.